packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1484,6 +1485,39 @@ abstract class FlutterCommand extends Command<void> {
     ]);
   }
 
+  void _addFeatureFlagsToDartDefines(List<String> dartDefines) {
+    if (dartDefines.any((String define) => define.startsWith(kEnabledFeatureFlags))) {
+      throwToolExit(
+        '$kEnabledFeatureFlags is used by the framework and cannot be '
+        'set using --${FlutterOptions.kDartDefinesOption} or --${FlutterOptions.kDartDefineFromFileOption}.\n'
+        '\n'
+        'Use the "flutter config" command to enable feature flags.',
+      );
+    }
+
+    final StringBuffer enabledFeatureFlags = StringBuffer();
+    for (final Feature feature in featureFlags.allFeatures) {
+      if (!featureFlags.isEnabled(feature)) {
+        continue;
+      }
+
+      final String? runtimeId = feature.runtimeId;
+      if (runtimeId == null) {
+        continue;
+      }
+
+      if (enabledFeatureFlags.isNotEmpty) {
+        enabledFeatureFlags.write(',');
+      }
+
+      enabledFeatureFlags.write(runtimeId);
```

Comment:
Any particular reason we're using a `StringBuffer` here rather than storing the `runtimeId`s in a `List<String>` and using `runtimeIds.join(',')` to build the string? 

Something like this would be more concise:

```dart
final String enabledFeatures = featureFlags.allFeatures
    .where((Feature f) => featureFlags.isEnabled(f) && f.runtimeId != null)
    .map((Feature f) => f.runtimeId!)
    .join(',');

```

Reply:
Ooh love it, great suggestion. Will update!

Reply:
Updated!